src/main/java/com/powsybl/sld/server/SingleLineDiagramController.java

```diff
@@ -280,27 +273,11 @@ public ResponseEntity<Collection<String>> getAvailableSvgComponentLibraries() {
             @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId,
             @Parameter(description = "depth") @RequestParam(name = "depth", required = false) int depth,
             @Parameter(description = "Initialize NAD with Geographical Data") @RequestParam(name = "withGeoData", defaultValue = "true") boolean withGeoData) throws InterruptedException, ExecutionException {
-        CompletableFuture<String> futureNAD = CompletableFuture.supplyAsync(() -> {
-            try {
-                LOGGER.debug("getNetworkAreaDiagramSvg request received with parameter networkUuid = {}, voltageLevelsIds = {}, depth = {}", networkUuid, sanitizeParam(voltageLevelsIds.toString()), depth);
-                SvgAndMetadata svgAndMetadata = networkAreaDiagramService.generateNetworkAreaDiagramSvg(networkUuid, variantId, voltageLevelsIds, depth, withGeoData);
-                String svg = svgAndMetadata.getSvg();
-                String metadata = svgAndMetadata.getMetadata();
-                Object additionalMetadata = svgAndMetadata.getAdditionalMetadata();
-                return OBJECT_MAPPER.writeValueAsString(
-                        OBJECT_MAPPER.createObjectNode()
-                                .put(SVG_TAG, svg)
-                                .putRawValue(METADATA, new RawValue(metadata))
-                                .putPOJO(ADDITIONAL_METADATA, additionalMetadata));
-            } catch (JsonProcessingException e) {
-                throw new PowsyblException("Failed to parse JSON response", e);
-            }
-        }, executorService);
-        return futureNAD.get();
-    }
-
-    @PreDestroy
-    private void preDestroy() {
-        executorService.shutdown();
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("getNetworkAreaDiagramSvg request received with parameter networkUuid = {}, voltageLevelsIds = {}, depth = {}", networkUuid, sanitizeParam(voltageLevelsIds.toString()), depth);
+        }
+        return diagramExecutionService
+            .supplyAsync(() -> networkAreaDiagramService.getNetworkAreaDiagramSvg(networkUuid, variantId, voltageLevelsIds, depth, withGeoData))
+            .get();
```

Comment:
this could simplify to
```suggestion
        return networkAreaDiagramService.getNetworkAreaDiagramSvg(networkUuid, variantId, voltageLevelsIds, depth, withGeoData));
```
?

Reply:
If we want it to be used for SLD and NAD, it makes sense to have it here. But from the code, it looks specialized to NAD.

Reply:
Done